paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/StripeIntentValidator.kt

```diff
@@ -26,25 +26,26 @@ internal class StripeIntentValidator @Inject constructor() {
             }
             stripeIntent is PaymentIntent &&
                 (
-                    (stripeIntent.status != StripeIntent.Status.RequiresPaymentMethod) &&
-                        (stripeIntent.status != StripeIntent.Status.RequiresAction)
+                    (stripeIntent.status == StripeIntent.Status.Canceled) ||
+                        (stripeIntent.status == StripeIntent.Status.Succeeded) ||
+                        (stripeIntent.status == StripeIntent.Status.RequiresCapture)
                     ) -> {
                 error(
                     """
-                        A PaymentIntent with status='requires_payment_method' or 'requires_action` is required.
+                        PaymentSheet received a PaymentIntent in a terminal state.
```

Comment:
lurking, hope that is ok! we should return a clearer more actionable error message here as some users might get confused on what a "terminal state" means. So maybe something like `PaymentSheet cannot set up a PaymentIntent in status: {stripeIntent.status}` or something like that.

Reply:
thanks for the tip, updated